### PR TITLE
[feat] Add Storage browser download tests

### PR DESCRIPTION
**Pull request by ashika112**
#### Description of changes

- Adds download e2e test for storage browser

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/e2e/cypress/integration/common/storagebrowser.ts [hunk 1]
```diff
@@ -34,3 +34,34 @@ When(
     });
   }
 );
+
+Then('I see download for file {string}', (name: string) => {
+  cy.contains('table tbody td:nth-child(2)', new RegExp('^' + name + '$'))
+    .siblings()
+    .last()
+    .children('button');
```
Comment (jordanvn) [inline packages/e2e/cypress/integration/common/storagebrowser.ts:42]: Is this the simplest way to locate this download button?

Comment (ashika112) [reply packages/e2e/cypress/integration/common/storagebrowser.ts:42]: Unfortunately yes. This kinda also checks you are clicking download of that particular row

File: packages/e2e/cypress/integration/common/storagebrowser.ts [hunk 2]
```diff
@@ -34,3 +34,34 @@ When(
...
+  cy.contains('table tbody td:nth-child(2)', new RegExp('^' + name + '$'))
+    .siblings()
+    .last()
+    .children('div');
+});
+
+Then('I click and see download succeed for {string}', (name: string) => {
+  cy.intercept('HEAD', 'https://*.s3.*.amazonaws.com/**').as(
+    'downloadValidation'
+  );
```
Comment (jordanvn) [inline packages/e2e/cypress/integration/common/storagebrowser.ts:55]: Nice 🚀

File: packages/e2e/features/ui/components/storage/storage-browser/download.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  Background:
+    Given I'm running the example "ui/components/storage/storage-browser/default-auth"
+
+  @react
+  Scenario: Download is available for files
+    When I type my "email" with status "CONFIRMED"
+    Then I type my password
+    Then I click the "Sign in" button
+    When I click the first button containing "public"
+    Then I see download for file "001_dont_delete_file.txt"
```
Comment (ashwinkumar6) [inline packages/e2e/features/ui/components/storage/storage-browser/download.feature:12]: nit:
Should we move this file inside `DO_NOT_DELETE` folder. That way all required items for e2e are in one place

**Review by ashwinkumar6 — APPROVED** [review]

**Review by jordanvn — APPROVED** [review]

File: packages/e2e/features/ui/components/storage/storage-browser/download.feature (hunk 3 above)
Comment (jordanvn) [reply packages/e2e/features/ui/components/storage/storage-browser/download.feature:12]: +1 to this. Should do a follow-up to reorganize data for testing

**Pull request merged**